### PR TITLE
Add command and event listener for sending a text message through datachannels

### PR DESCRIPTION
**Pull request by horymury**
@saghul

### CODE REVIEW

File: modules/API/external/external_api.js [hunk 1]
```diff
@@ -75,7 +76,8 @@ const events = {
     'dominant-speaker-changed': 'dominantSpeakerChanged',
     'subject-change': 'subjectChange',
     'suspend-detected': 'suspendDetected',
-    'tile-view-changed': 'tileViewChanged'
+    'tile-view-changed': 'tileViewChanged',
+    'endpoint-text-message-received': 'endpointTextMessageReceived'
```
Comment (saghul) [inline modules/API/external/external_api.js]: please sort this alphabetically

File: modules/API/external/external_api.js [hunk 2]
```diff
@@ -40,7 +40,8 @@ const commands = {
     toggleFilmStrip: 'toggle-film-strip',
     toggleShareScreen: 'toggle-share-screen',
     toggleTileView: 'toggle-tile-view',
-    toggleVideo: 'toggle-video'
+    toggleVideo: 'toggle-video',
+    sendEndpointTextMessage: 'send-endpoint-text-message'
```
Comment (saghul) [inline modules/API/external/external_api.js]: please sort these alphabeticcally

File: conference.js [hunk 3]
```diff
@@ -2041,7 +2042,18 @@ export default {
 
         room.on(
             JitsiConferenceEvents.ENDPOINT_MESSAGE_RECEIVED,
-            (...args) => APP.store.dispatch(endpointMessageReceived(...args)));
+            (...args) => {
+                APP.store.dispatch(endpointMessageReceived(...args));
+                if (args && args.length >= 2 && args[1].name === ENDPOINT_TEXT_MESSAGE_NAME) {
```
Comment (saghul) [inline conference.js]: can you extract this to objects to make it more readable? like `particcipant` and `data` for example.

File: conference.js [hunk 4]
```diff
@@ -2041,7 +2042,23 @@ export default {
 
         room.on(
             JitsiConferenceEvents.ENDPOINT_MESSAGE_RECEIVED,
-            (...args) => APP.store.dispatch(endpointMessageReceived(...args)));
+            (...args) => {
+                APP.store.dispatch(endpointMessageReceived(...args));
+                if (args && args.length >= 2) {
+                    const sender = args[0];
+                    const eventData = args[1];
```
Comment (saghul) [inline conference.js]: You can simplify this: `const [sender, eventData] = args;`

File: modules/API/API.js [hunk 5]
```diff
@@ -155,6 +155,18 @@ function initCommands() {
         'avatar-url': avatarUrl => {
             sendAnalytics(createApiEvent('avatar.url.changed'));
             APP.conference.changeLocalAvatarUrl(avatarUrl);
+        },
+        'send-endpoint-text-message': (to, text) => {
+            sendAnalytics(createApiEvent('send.endpoint.message'));
```
Comment (saghul) [inline modules/API/API.js]: If large volumes of data are going to be sent with this mechanism, sending an analytics event for each message seems excessive

Comment (horymury) [reply modules/API/API.js]: Removed the analytics sending. Will ask today on standup if it s needed or not.

**Pull request merged**